### ISSUE TITLE
dart_eval runtime exception: type 'bool' is not a subtype of type '$Instance' in type cast when using || operator

### ISSUE THREAD

**Issue by Noobware1** [ethanblake4/dart_eval#142]
code
```
  VideoSource toVideoSource(Map j, bool backup) {
    final labelStr = j['labelStr'];
    final url = j['file'];
    if (labelStr != "hls P" || labelStr != "auto P") {
      return VideoSource(
        url: url,
        format: VideoFormat.other,
        isBackup: backup,
      );
    } else {
      return VideoSource(
        url: url,
        format: VideoFormat.hls,
        isBackup: backup,
      );
    }
  }
```
error
```
dart_eval runtime exception: type 'bool' is not a subtype of type '$Instance' in type cast
#0      InvokeDynamic.run (package:dart_eval/src/eval/runtime/ops/objects.dart:99:31)
#1      Runtime.bridgeCall (package:dart_eval/src/eval/runtime/runtime.dart:847:12)
#2      EvalStaticFunctionPtr.call (package:dart_eval/src/eval/runtime/function.dart:86:13)
at GogoCDN.toVideoSource()
at <asynchronous gap>

RUNTIME STATE
=============
Program offset: 1330
Stack sample: [L0: Instance of '$ExtractorApi', L1: Instance of '$Completer<dynamic>', L2: Instance of '$ExtractorLink', L3: $"https://goone.pro/streaming.php?id=MTE2ODI1&title=Kimetsu+no+Yaiba+Episode+1", L4: $"GET", L5: null, L6: Instance of '$Future<dynamic>', L7: Instance of '$OkHttpResponseObject', L8: Instance of '$DocumentObject', L9: $"script[data-name="episode"]"]
Args sample: [L0: true]
Call stack: [0, -1]
TRACE:
1324: PushConstant (C111)
1325: BoxString (L10)
1326: CheckEq (L4 == L10)
1327: PushReturnValue ()
1328: LogicalNot (L11)
1329: PushArg (L12)
1330: InvokeDynamic (L9.C112)  <<< EXCEPTION
1331: PushReturnValue ()
1332: Unbox (L13)
1333: JumpIfFalse (@1349 if L13 == false)

#0      Runtime.bridgeCall (package:dart_eval/src/eval/runtime/runtime.dart:855:7)
#1      Await._suspend (package:dart_eval/src/eval/runtime/ops/bridge.dart:169:13)
<asynchronous suspension>
```

**Comment by Noobware1** [comment]
example classes and yes they are bridged

```
class VideoSource {
final String url;
final int? quality
final VideoFormat format 
final bool isBackup


  VideoSource({required this.url,  this.quality, required this.format,  this.isBackup = false});
}

enum  VideoFormat {
  hls,
  mp4,
  other,
}
```

**Comment by Noobware1** [comment]
Fixed in v0.7.4
